### PR TITLE
Add hardware reboot cause as actual reboot cause when soft reboot fails

### PR DESCRIPTION
**Pull request by sujinmkang**
Why I did it
Add the hardware reboot cause when the previous software reboot failed

How I did it
Check both hardware reboot cause and software reboot cause.
Add the hardware reboot as actual reboot cause
if any hardware reboot cause is available for any software reboot.

How to verify it
Perform fast/warm/soft-reboot, cold reboot and hardware reboots and check the reboot-cause

### CODE REVIEW

**Comment by prgeor** [comment]
@sujinmkang can you test this on Dell Z9332F
sync; sleep 2; sync; echo -c /proc/sysrq-trigger

See if you see the correct reboot cause

File: scripts/determine-reboot-cause [hunk 1]
```diff
@@ -122,7 +122,12 @@ def find_hardware_reboot_cause():
     else:
         sonic_logger.log_info("No reboot cause found from platform api")
 
-    hardware_reboot_cause = "{} ({})".format(hardware_reboot_cause_major, hardware_reboot_cause_minor)
+    hardware_reboot_cause_minor_str = ""
+    if hardware_reboot_cause_minor:
+        hardware_reboot_cause_minor_str = " ({})".format(hardware_reboot_cause_minor)
+
+    hardware_reboot_cause = hardware_reboot_cause_major + hardware_reboot_cause_minor_str
```
Comment (prgeor) [inline scripts/determine-reboot-cause:129]: can we add space between the two strings ?

File: scripts/determine-reboot-cause [hunk 2]
```diff
@@ -158,6 +163,50 @@ def get_reboot_cause_dict(previous_reboot_cause, comment, gen_time):
...
+        previous_reboot_cause = hardware_reboot_cause
+        # Check if any software reboot was issued before this hardware reboot happened
+        if software_reboot_cause is not REBOOT_CAUSE_UNKNOWN:
+            additional_reboot_info = software_reboot_cause
+        elif proc_cmdline_reboot_cause is not None:
+            additional_reboot_info = proc_cmdline_reboot_cause
+    elif proc_cmdline_reboot_cause is not None:
+        if software_reboot_cause is not REBOOT_CAUSE_UNKNOWN:
+            # Get the reboot cause from REBOOT_CAUSE_FILE
+            previous_reboot_cause = software_reboot_cause
```
Comment (prgeor) [inline scripts/determine-reboot-cause:202]: ignoring the proc_cmdline_reboot_cause which is what being checked at line 199?

File: scripts/determine-reboot-cause (hunk 1 above)
Comment (sujinmkang) [reply scripts/determine-reboot-cause:129]: It's added when hardware_reboot_cause_minor_str was constructed.

File: scripts/determine-reboot-cause (hunk 2 above)
Comment (sujinmkang) [reply scripts/determine-reboot-cause:202]: As I explained, this is for the fast/warm-reboot case. the software_reboot_cause contains the user and time information.

**Review by prgeor — APPROVED** [review]

**Comment by prgeor** [comment]
@sujinmkang please make sure this is ported to 202205

**Pull request merged**